### PR TITLE
Add locale map from source Id bean

### PR DESCRIPTION
**Pull request by phuongnq**

### CODE REVIEW

**Comment by phuongnq** [comment]
Sample REST script URL: http://localhost:8080/api/plugins/control/custom-locale/locale-map.json?sourceId=794ff24c-039f-a778-84ea-8a97a1908fd0
Sample result:

```
{
    "en": "/site/website/en/articles/2016/12/top-books-for-young-women/index.xml",
    "fr": "/site/website/fr/articles/2021/05/top-books-for-young-women-c7be04cd/index.xml"
}
```

File: scripts/classes/plugins/control/customlocale/LocaleHelper.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    LocaleHelper(elasticsearch) {
+      this.elasticsearch = elasticsearch
+      this.sourceIdFieldName = DEFAULT_SOURCE_ID_FIELD_NAME
+      this.codeFieldName = DEFAULT_CODE_FIELD_NAME
+    }
+
+    def searchSourceId(sourceId) {
+        def result = [:]
+
+        if (!sourceId) return result
```
Comment (avasquez614) [inline scripts/classes/plugins/control/customlocale/LocaleHelper.groovy]: Please use:

```
if (!sourceId) {
}
```

We by convention don't like `if` one-liners.

File: scripts/rest/plugins/control/custom-locale/locale-map.get.groovy [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import plugins.control.customlocale.LocaleHelper
+
+def sourceId = params.sourceId
+
+if (!sourceId) return [:]
```
Comment (avasquez614) [inline scripts/rest/plugins/control/custom-locale/locale-map.get.groovy]: Please use:

```
if (!sourceId) {
}
```

We by convention don't like `if` one-liners.

**Review by avasquez614 — CHANGES_REQUESTED** [review]

File: scripts/classes/plugins/control/customlocale/LocaleHelper.groovy (hunk 1 above)
Comment (phuongnq) [reply scripts/classes/plugins/control/customlocale/LocaleHelper.groovy]: Updated to use {}

File: scripts/rest/plugins/control/custom-locale/locale-map.get.groovy (hunk 2 above)
Comment (phuongnq) [reply scripts/rest/plugins/control/custom-locale/locale-map.get.groovy]: Updated to use {}

**Comment by phuongnq** [comment]
ported to https://github.com/avasquez614/wwwpokemoncom/pull/2

**Pull request closed without merging**